### DISCUSSION TITLE
Running default cargo new

### DISCUSSION THREAD

**Discussion by KarolStola** [cargo-generate/cargo-generate#802] [Q&A] [answered]
Hi!

I want to create a template that just adds some files to the standard `cargo new` command. My first intuition is to use hooks to write a rhai script calling that, but I couldn't find any info on how to achieve it in it's documentation (I don't know rhai at all). Is there any other way to do that?

**Comment by taurr** [comment, marked as answer]
if you want to run `cargo new` as part of the template, it can be done using `Rhai` as you describe, but its a bit cumbersome...

Command line:
```shell
cargo generate --allow-commands <template>
```

Then you need the actual template with the files you want to add, as well as a `Rhai` script that does:
```rhai
let output = system::command("cargo", ["new", "additinal args..."]);
```

NOTE: if the external command (`cargo new` in this case) requires **any** kind of input it **will fail**! This is by design, and is a current known limitation.

... if it was me, considering that `cargo new` doesn't give you much to start with, I wouldn't bother! Just add the few files including the `cargo new` output to a small template and call it a day. Running `cargo` from within the template is still a security issue (hence the need for the `--allow-commands` flag), and should really be avoided unless absolutely required - and in this case the external command does not give you any advantage to just add the files to the actual template.
